web3/abi_serialization.nim

```diff
@@ -0,0 +1,39 @@
+
+import serialization
+import faststreams
+
+{.push raises: [].}
+
+serializationFormat Abi,
+                   mimeType = "application/ethereum‑abi"
+
+type
+  AbiReader* = ref object
```

Comment:
```suggestion
  AbiReader* = object
```

these don't need to be ref since all they do is provide a typed wrapper effectively and semantically shouldn't be shared (which ref implies).

removing the ref should give a tiny tiny bump in your benchmark as well and make them safer to use (one less nil to worry about).

we also tend to make all functions take `var XxxReader`/`var XxxWriter` in the serialization framework world